CodePush.js

```diff
@@ -41,7 +40,7 @@ async function checkForUpdate(deploymentKey = null) {
    * in any updates for current app store version, regardless of hash.
    */
   let queryPackage;
-  if (localPackage && localPackage.appVersion && semver.compare(localPackage.appVersion, config.appVersion) === 0) {
+  if (localPackage && localPackage.appVersion && localPackage.appVersion === config.appVersion) {
```

Comment:
I just quickly looked at the iOS implementation, and we already know the update and binary version, so it would be super simple to change the logic to clear updates in debug mode is the versions are different. That way, our assumption is that if you're not bumping the version in debug mode, we keep old updates so you don't get unnecessary CodePush updates, but as soon as you bump the version, even in debug mode, we clear updates. We actually need to do this anything to accommodate the new range feature, since the newly bumped version may no longer be eligible for the locally installed updates.